m3u8/model.py

```diff
@@ -174,6 +174,9 @@ def _initialize_attributes(self):
         for attr, param in self.simple_attributes:
             setattr(self, attr, self.data.get(param))
 
+        for i, segment in enumerate(self.segments, self.media_sequence or 0):
```

Comment:
I learned something new here :) I didn't know `enumerate` would work with two sources :)

Just out of curiosity, why does it return first the number iterator and secondly the list iterator? (since they were passed in other order)

Reply:
Two tricks here.

First, enumerate has a two argument form: `iterable`, `start_number`. For us, `self.segments` is the iterable, since the `SegmentList` type can be iterated. We want to start counting from the media sequence number, so we use it as the second argument.

Second, the `self.media_sequence or 0` trick. I want to handle the case where `media_sequence` is `None` and treat that ias `0`, as per the HLS spec. The `or` operator takes the left argument if it's truthy (`None` isn't) and the right operator if it's not. 

```python
None or 0  # Picks 0
0 or 0  # Picks the second 0
1 or 0  # Picks 1
2 or 0  # Picks 2
# etc
```

Reply:
cool ty